metis-pattern-analysis/src/main/java/eu/europeana/patternanalysis/ProblemPatternAnalyzer.java

```diff
@@ -53,51 +71,176 @@ public List<ProblemPattern> analyzeRecord(String rdfString) throws Serialization
+            >= LCS_CALCULATION_THRESHOLD;
+    final Predicate<String> distancePredicate = description -> Math.abs(title.length() - description.length())
+        <= TITLE_DESCRIPTION_LENGTH_DISTANCE;
+    return descriptions.stream().filter(StringUtils::isNotBlank).filter(not(title::equalsIgnoreCase))
+                       .filter(lcsPredicate.and(distancePredicate)).collect(toList());
+  }
+
+  /**
+   * Check whether a title is not human-readable.
+   * <p>
+   * We check this by:
+   *   <ul>
+   *     <li>Whether there are more than 5 characters that are not valid.
+   *     Non valid characters are considered characters that are not alphanumeric and are not simple "literal" spaces(tabs,
+   *     new lines etc. are considered invalid characters).
+   *     This is performed with regex unicode matching {@link #UNRECOGNIZABLE_CHARACTERS_REGEX} and should support all languages.
+   *     For more information check <a href="https://www.regular-expressions.info/unicode.html#category">unicode regex</a></li>
+   *     <li>The title does not fully contain an identifier</li>
+   *   </ul>
+   * </p>
+   *
+   * @param titles the list of titles
+   * @param identifiers the list of identifiers
+   * @return the list of problem occurrences encountered
+   */
+  private List<ProblemOccurrence> checkP5(List<String> titles, List<String> identifiers) {
+    final Predicate<String> moreThanThresholdUnrecognizableCharacters = s ->
+        UNRECOGNIZABLE_CHARACTERS_PATTERN.matcher(s).results().count() > UNRECOGNIZABLE_CHARACTERS_THRESHOLD;
+    final Predicate<String> containsIdentifier = s -> identifiers.stream().anyMatch(s::contains);
+    return titles.stream().filter(moreThanThresholdUnrecognizableCharacters.or(containsIdentifier))
+                 .map(title -> new ProblemOccurrence(format("Unrecognized title: %s", title))
+                 ).collect(toList());
+  }
+
+  /**
+   * Check whether the record has titles of {@link #MIN_TITLE_LENGTH} characters or fewer.
+   *
+   * @param titles the list of titles
+   * @return the list of problem occurrences encountered
+   */
   private List<ProblemOccurrence> checkP6(List<String> titles) {
     return titles.stream().filter(title -> title.length() <= MIN_TITLE_LENGTH)
-                 .map(title -> new ProblemOccurrence(format("Non meaningful title: %s", title))).collect(Collectors.toList());
+                 .map(title -> new ProblemOccurrence(format("Non meaningful title: %s", title))).collect(toList());
+  }
+
+  /**
+   * Check whether the record is lacking a description (or only has empty descriptions).
+   *
+   * @param descriptions the list of descriptions
+   * @return the list of problem occurrences encountered
+   */
+  private List<ProblemOccurrence> checkP7(List<String> descriptions) {
+    if (CollectionUtils.isEmpty(descriptions) || descriptions.stream().allMatch(StringUtils::isBlank)) {
```

Comment:
:heavy_check_mark: I think this is related with the comments I made with ```getChoicesInStringList``` what hapens if there are ```nulls``` is this possible to happen?

Reply:
This checks for null items already with StringUtils::isBlank